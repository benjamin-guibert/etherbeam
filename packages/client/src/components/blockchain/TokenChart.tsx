import React, { FC } from 'react'
import TradingViewWidget, { Themes } from 'react-tradingview-widget'

interface TokenChartProps {
  symbol: string
  width?: number
  height?: number
}

const TokenChart: FC<TokenChartProps> = ({ symbol, width, height }) => {
  const classNames = ['my-shadow', (!width && 'w-100') || '', (!height && 'h-100') || ''].join(' ')
  const style = {
    width: !!width ? `${width}px` : null,
    height: !!height ? `${height}px` : null,
  }

  return (
    <div className={classNames} style={style}>
      <TradingViewWidget
        symbol={`UNISWAP:${symbol}WETH`}
        interval={60}
        timezone="Europe/Paris"
        theme={Themes.DARK}
        allow_symbol_change={false}
        show_popup_button
        withdateranges
        autosize
      />
    </div>
  )
}

export default TokenChart
