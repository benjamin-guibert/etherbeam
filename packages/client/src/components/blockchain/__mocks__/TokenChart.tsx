import React, { ReactElement } from 'react'

const TokenChart = (): ReactElement => {
  return <div></div>
}

export default TokenChart
