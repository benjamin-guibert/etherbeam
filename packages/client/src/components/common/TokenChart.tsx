import React, { ReactElement } from 'react'
import TradingViewWidget, { Themes } from 'react-tradingview-widget'

interface TokenChartProps {
  symbol: string
  width?: number
  height?: number
}

const TokenChart = ({ symbol, width, height }: TokenChartProps): ReactElement => {
  const classNames = [(!width && 'w-100') || '', (!height && 'h-100') || ''].join(' ')
  const style = {
    width: !!width ? `${width}px` : null,
    height: !!height ? `${height}px` : null,
  }

  return (
    <div className={classNames} style={style}>
      <TradingViewWidget
        symbol={`UNISWAP:${symbol}WETH`}
        interval={60}
        timezone="Europe/Paris"
        theme={Themes.DARK}
        allow_symbol_change={false}
        show_popup_button
        withdateranges
        autosize
      />
    </div>
  )
}

export default TokenChart
