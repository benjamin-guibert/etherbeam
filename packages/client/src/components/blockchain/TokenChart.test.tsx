import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'

jest.mock('react-tradingview-widget', () => {
  return {
    __esModule: true,
    default: () => {
      return <article></article>
    },
    Themes: {
      DARK: 'DARK',
    },
  }
})

import TokenChart from './TokenChart'

describe('<TokenChart />', () => {
  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<TokenChart symbol="UNI" />)
    })

    it('should render div', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('w-100')).toBeTruthy()
      expect(div.hasClass('h-100')).toBeTruthy()
    })

    it('should render widget', () => {
      const widget = component.find('div').at(0).children().at(0)
      expect(widget.prop('symbol')).toBe('UNISWAP:UNIWETH')
      expect(widget.prop('interval')).toBe(60)
      expect(widget.prop('timezone')).toBe('Europe/Paris')
      expect(widget.prop('allow_symbol_change')).toBeFalsy()
      expect(widget.prop('show_popup_button')).toBeTruthy()
      expect(widget.prop('withdateranges')).toBeTruthy()
      expect(widget.prop('autosize')).toBeTruthy()
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('With width', () => {
    beforeAll(() => {
      component = shallow(<TokenChart symbol="UNI" width={200} />)
    })

    it('should render div', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('w-100')).toBeFalsy()
      expect(div.hasClass('h-100')).toBeTruthy()
    })
  })

  describe('With height', () => {
    beforeAll(() => {
      component = shallow(<TokenChart symbol="UNI" height={200} />)
    })

    it('should render div', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('w-100')).toBeTruthy()
      expect(div.hasClass('h-100')).toBeFalsy()
    })
  })

  describe('With width', () => {
    beforeAll(() => {
      component = shallow(<TokenChart symbol="UNI" width={200} height={200} />)
    })

    it('should render div', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('w-100')).toBeFalsy()
      expect(div.hasClass('h-100')).toBeFalsy()
    })
  })
})
