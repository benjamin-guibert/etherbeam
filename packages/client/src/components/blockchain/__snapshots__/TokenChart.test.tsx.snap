// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokenChart /> Default should match snapshot 1`] = `
<div
  class="my-shadow w-100 h-100"
>
  <article />
</div>
`;
